### DISCUSSION TITLE
nordlynx slower than nordvpn?

### DISCUSSION THREAD

**Discussion by LeBrunton** [bubuntux/nordlynx#17] [Q&A]
I was previously using your nordvpn package to connect to my vpn, and downloading torrents using Transmission was fast and consistent. As of last week nordvpn would no longer connect so I moved to nordlynx, but now downloading torrents in the same manner is a lot slower, but also the speeds keep fluctuating, more than I would expect.

As a test I shut down the vpn and downloaded the torrents through my normal connection using Transmission, now of course it will be faster as there is no vpn involved, but the speed fluctuation disappeared :/

Is anyone else seeing this?

Lewis

**Comment by rodrigoi** [comment]
Having the same problems. Performance is abysmal compared to the NordVPN container. I've tried several filters (countries, server groups) and nothing works.

**Reply by LeBrunton** [reply]
Glad I'm not going mad

**Reply by rodrigoi** [reply]
I've also tried switching from Transmission to qBittorrent, just in case, but that didn't help either. Transmission is the only service using the VPN, and everything else works (VPN split tunnel, filters, etc).

**Reply by bubuntux** [reply]
In theory this container should be faster than the nordvpn one, in my end nordlynx is lighter and way faster, but your mileage may variate.

**Reply by Crooklar** [reply]
after a few months of messing around perodically with different settings, this container/method is 100% slower.

On the same machine, this container vs the old continer, using nordlynx as the 'technology', this container/method is much slower.

**Reply by ginodesilva** [reply]
Just curious. What metrics do you compare? Max download of torrents? Newsgroups? Speedtest?

I've never used the NordVPN container so can't compare but am getting max download for my connection through the vpn.

**Reply by Crooklar** [reply]
Previously I could download the most popular ISO torrents at the full ISP speed.

Under the new container, I can get 2 Mb/s max which is 1/4 of the speed i would expect and am capable of downloading at.

If I used qbittorrent on desktop, I’d get 8Mb/s.

**Reply by ginodesilva** [reply]
Is the number of trackers you can reach roughly the same? What distro are you running nordlynx on? What's your compose?

**Reply by bubuntux** [reply]
@Crooklar  getting ISP speed using a VPN seems very unlikely, that may be a sign that the vpn is not working, nordlynx is faster no doubt

**Reply by sn4201** [reply]
@bubuntux I am having the exact same issue as OP.  If i use Gluetun OpenVPN container with NordVPN on my raspberry pi, i hit ~9mb/s download before the CPU on my Rpi is maxed out while downloading (i'm assuming because the Rpi doesn't have hardware accelerated AES decoding?)

Switching to nordlynx container (tested with multiple different servers) my download speed maxes around 1.5-2mb/s.  Doesnt max out my CPU at all, so it doesn't seem to be hardware-limited like the openvpn client.  Also when I connect to nordlynx via nordvpn linux client on my laptop and test torrent speeds, i easily reach 20mb/s+ download speeds.  No clear reason why this nordlynx container can't go any faster....Hope you can look into this further.

**Comment by tmknight** [comment]
For me, speed was good, however, inconsistent and unstable - frequently "unhealthy", slow DNS lookups and significant delay in page loads and even crashing my host VM.  Went back to NordVPN container...although it is stated as no longer being maintained...much more stable and consistent performance and faster than OpenVPN...that's a win in my book.  A shame that @bubuntux won't be maintaining the native client.

**Reply by sn4201** [reply]
Update to anyone who may be interested, I have managed to use the linuxserver.io group's wireguard container and created my own wg0.conf file to use with NordVPN servers.  This works much faster and with fewer resources than openvpn-based nordvpn containers, and is working much better than this @bubuntux nordlynx container in its current form.

I will see if in the future i can write up a straightforward and concise guide on how to create a wg0.conf file, although there are instructions out there already if you look.

**Reply by bubuntux** [reply]
@sn4201  this image is based on linuxserver.io wireguard just parametrize the nordvpn creds basically, you may get stall connection using regular linuxserver.io container after 4 hours or so, something that is addressed in nordlynx container, but please if linuxserver.io works without issues for more than a few days report back, the less work i have to do (maintaining this image) the better

**Reply by bubuntux** [reply]
@tmknight  if openVPN works for you, use that, the native client is a pain to work with, feel free to fork the image and maintain it yourself or just use other options like Gluten

**Reply by tmknight** [reply]
> ...the native client is a pain to work with...

@bubuntux can you elaborate on the issues you've encountered with the native client?

**Reply by bubuntux** [reply]
@tmknight  is just a black box, they mess with iptables and more that is hard to mess around, also they deprecated auth options and is just a pain keeping it up-to-date, i rather this nordlyx approach, this container just create the regular wirgeuard config file using the nord api and use regular and plain wireguard

**Comment by alongwhile** [comment]
For those having performance issues with this nordlynx container and until it can be properly fixed, I threw together a how-to guide [here](https://github.com/alongwhile/nordvpn-wg) which walks you through creating your own wireguard configuration file so you can use it with any vanilla wireguard client.  Also included specific instructions for setting up a simple wireguard docker client.  I was able to drop this into my existing docker stacks in place of @bubuntux nordlynx container and it resolved all of my performance issues.  Hope it can be useful to others!  @tmknight Have a look and let me know what you think!

**Reply by tmknight** [reply]
@alongwhile, I've read through your md; ultimately, bubuntux is doing the same thing, though automated via his scripts (e.g. run).  I surmise your performance improvement may be down to your inclusion of modules that may not be part of this Nordlynx image...i.e. you basing from linuxserver.io WG container and this from baseimage-alpine and installing WG.

This can be tested by pulling the repo, then changing the [dockerfile ](https://github.com/bubuntux/nordlynx/blob/main/Dockerfile) to lscr.io/linuxserver/wireguard:latest, possibly a change to where config is stored (I haven't dissected), build, run.

**Reply by bubuntux** [reply]
yeah seems like it does the same as this container, one small difference is that the reconnect is not in place, so you can set the variable RECONNECT to empty `RECONNECT= ` to prevent the reconnect, wondering if that is degrading the performance somehow

**Reply by jgosnell56** [reply]
> yeah seems like it does the same as this container, one small difference is that the reconnect is not in place, so you can set the variable RECONNECT to empty `RECONNECT= ` to prevent the reconnect, wondering if that is degrading the performance somehow

I think having the RECONNECT variable set definitely hurst performance.  I went from ~15Mbps downloads to ~75Mbps down after removing that variable from my Unraid template and rebuilding.

**Reply by bubuntux** [reply]
by removing the variable you restarted the container and refreshed the connection, which is the function of that variable, the connection just degrade over time, that is just nordvpn issues, not this container

**Reply by jgosnell56** [reply]
Negative.  I have a script running to restart the container every 2 hours due to degradation.  Even after those restarts with the variable configured, I never got above 15Mbps or so.  Immediately after removing that variable and restarting, I hit >75Mbps and my upload speeds now seem to be better.  I came here looking to see if others were having performance issues after time, or if others had to restart the container as much as I was having to.  Maybe this is just coincidental, but hard to see that as much as I rebooted it previously.

**Reply by jgosnell56** [reply]
Side issue, the icon URL for the container is no longer valid.
